### PR TITLE
consider ambiwidth for border

### LINKED ISSUE #1856: set ambiwidth=double breaks popup border layouts
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
  - v0.20.0
- [x] I have searched through the existing issues

## Info

- OS
    - [ ] Linux
    - [x] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [x] bash
    - [x] zsh
    - [ ] fish
    
## Problem / Steps to reproduce

```
:let g:fzf_layout = { 'window': { 'width': 0.6, 'height': 0.6, 'rounded': v:false } }
:set ambiwidth=double
:Files
```

068a6322433b1a7acf5ced08a92ea04b

### PR DESCRIPTION
**Pull request by kyoh86**
fix: #1856

Both of this and #1857 will fix #1856,
but this one won't create new option "ascii".

### CODE REVIEW

File: plugin/fzf.vim [hunk 1]
```diff
@@ -860,16 +863,21 @@ function! s:popup(opts) abort
     let style = 'sharp'
   endif
 
+  let ambidouble = 1
+  if &ambiwidth == 'double'
+    let ambidouble = 2
+  endif
```
Comment (junegunn) [inline plugin/fzf.vim]: I would suggest simplifying this using ternary operator:

```vim
let ambidouble = &ambiwidth == 'double' ? 2 : 1
```

File: plugin/fzf.vim [hunk 2]
```diff
@@ -844,6 +844,9 @@ endif
 function! s:popup(opts) abort
   " Size and position
   let width = min([max([0, float2nr(&columns * a:opts.width)]), &columns])
+  if &ambiwidth == 'double'
```
Comment (junegunn) [inline plugin/fzf.vim]: Let's calculate `ambidouble` before this line and use it (because `x % 1` is always 0)

```vim
let width += width % ambidouble
```

Comment (kyoh86) [reply plugin/fzf.vim]: thanks

**Comment by kyoh86** [comment]
@junegunn thank you for your review!
I fixed.

**Comment by junegunn** [comment]
Merged, thanks!

**Pull request merged**